### DISCUSSION TITLE
how to press a virtual key (former homematic.virtualkey / PRESS_SHORT)

### DISCUSSION THREAD

**Discussion by AlexaMiller** [SukramJ/aiohomematic#215] [Q&A] [answered]
Servus,
i'm looking for the the service with which to press a virtual key,
has this been implemented yet - am i looking in the wrong place?

former definition was:
          action: call-service 
          service: homematic.virtualkey
          service_data:
            address: HmIP-RCV-1
            channel: 3
            param: PRESS_SHORT
            interface: HmIP_debmatic

Und eine prinzipielle Frage - wird hier eher deutsch oder englisch gesprochen?
Liebe Grüße, Alexa

**Comment by SukramJ** [comment]
Ich antworte so wie angefragt wird.
Den Service gibt es nicht mehr, aber du kannst in einer Automatisierung unter Aktion (und auch Trigger) die virtuellen Fernbedienungen als Device auswählen. Dann stehen die einzelnen Aktionen in der Auswahl zur Verfügung.

**Reply by Lemocin** [reply]
Kurze Frage dazu: Am Anfang hast du die ja als Buttons implementiert. Damit konnte man die Tasten bequem auf Lovelace einbinden. Gibt es so eine Möglichkeit immernoch oder muss ich das ganze dafür dann z.B. in ein Skript packen.

**Reply by AlexaMiller** [reply]
Die Implementierung als Button wäre eine schöne Lösung! +1

**Reply by SukramJ** [reply]
Dann gehen die Diskussionen wieder los, das soviel Entitäten erstellt werden.
Das sind dann **pro** virtueller Fernbedienung 100 Entitäten (50* Press_Short/ Press_Long).
Desweitern verwirrt das Ganze weil die Buttons auch als Action Trigger auswählbar sind, die dann aber nicht als Auslöser für den CCU-Remote Tastendruck dienen, sondern nur als HA Button Auslöser.

**Reply by Lemocin** [reply]
> Die Implementierung als Button wäre eine schöne Lösung! +1

Bis Version 0.15.0 gabs das als Buttons. Da gab es allerdings Gründe, das ganze zu Actions umzubauen. Die virtuellen Buttons drücke ich über die UI aktuell per Skript. Wenn man das Skript so erstellt, kann man es in der UI einbinden: 
![grafik](https://user-images.githubusercontent.com/4162496/149985626-b777e7a6-d105-4299-b457-ff3547f70276.png)

Ich habe mich nur gefragt ob es einen direkteren Weg gibt, den Tastendruck per UI auszulösen.

**Reply by SukramJ** [reply]
~~Schaut euch [das](https://www.home-assistant.io/integrations/button/) mal an. Ich denke, das ist für die paar Buttons die man braucht die bessere Lösung.~~ 
Blodsinn. Ihr habt ja keine Button.

**Reply by AlexaMiller** [reply]
Wäre schon nett ... könnte man die Buttons nicht auch evtl "deaktiviert" erstellen? Und User die das Feature möchten, aktivieren sie dann?
Rein egoistisch gedacht - ich verwende die Virtuellen Tasten ziemlich exzessiv - mir würde das somit das Erstellen von an die 70 Scripts ersparen :)

**Reply by SukramJ** [reply]
Drückst Du wirklich auf Knöpfe?
Was hast Du denn da für UseCases?

**Reply by Lemocin** [reply]
> Blodsinn. Ihr habt ja keine Button.

Hab mich grade schon gefragt, wie das hilft :D 



> Wäre schon nett ... könnte man die Buttons nicht auch evtl "deaktiviert" erstellen? Und User die das Feature möchten, aktivieren sie dann? 

Diese Diskussion wurde [schonmal geführt ](https://github.com/danielperna84/hahomematic/discussions/17)und ist auch leicht ausgeartet. Ich glaube nicht, dass das noch eine Lösung wäre...

**Reply by AlexaMiller** [reply]
UseCase zB: WinMatic lüften - eine virtuelle Taste pro WinMatic (15 Stk) definiert die das Gerät für n-Minuten lüftet. Eine Taste die sämtliche WinMatics schließt. Geschoßweise lüftet. Sämtliche Taster die für n-Minuten eine Aktion ausführen sollen (Garagentor, Dachbodentreppe, Einfahrt, ...). Virtuelle Aktionen die mit allen WRC6 im Haus verknüpft sind (Notruf etc)
Bewässerung. 28 Kreise die mittels virtuellen Tasten gesteuert werden.
Ja, wir haben hier eine wilde Installation :)

**Reply by SukramJ** [reply]
Interessant. Hab ihr eine Gärtnerei?

**Reply by AlexaMiller** [reply]
Nein. Nur einen großen Garten und zu viel Zeit :)
Nope ernsthaft .. Unser Garten ist etwas ungewöhnlich gestaltet und wir hatten die Möglichkeit. Sowas kommt halt raus wenn ein ISP-Betreiber und ein Programmierer ein Haus übernehmen.

**Reply by danielperna84** [reply]
@AlexaMiller 

Ohne mich jetzt zu tief da hinein zu denken: ist es nicht eigentlich Sinn der Hausautomation zu automatisieren? Also nicht 15 Mal individuell lüften, sondern in HA ein Script starten, welches dann der Reihe nach die Lüftung durchführt?

Ich habe keine WinMatics im Einsatz und weiß nicht wie die im UI bedienbar sind. Aber wenn es um's individuelle öffnen geht sollte doch das Entity im UI hierfür auch die Möglichkeit bieten. Da wäre der extra Knopf ja nur ein "Shortcut" für etwas was schon da ist. Eigentlich wäre doch ein extra Lovelace panel oder so mit den Fenstern in der Übersicht eine Lösung. Da kann man dann manuell lüften wie man Lust und Laune hat. 🤔

Oder anders gesagt: wenn man die virtuellen Knöpfe drücken will, dann liegt das "Problem" eigentlich woanders. Wobei das mit Direktverknüpfungen im Hintergrund ggf. auch wieder anders aussieht und ich es mir gerade vielleicht zu einfach mache. 😅

**Reply by AlexaMiller** [reply]
Die automatisierte Lüftung läuft über ein (im Moment NodeRED) Script, welches die Virtuellen Taster betätigt. Die Taster haben sich im Laufe der Jahre als praktisch erwiesen, da wir bereits einige User Interfaces durchlaufen haben (von Pocket Control über NodeRED Dashboard über Eigenbau bis zu aktuell Home Assistant). Diese Taster ersparen die Neuprogrammierung von "OG_Teekueche_WinMatic" jedesmal neu zu programmieren. 
Lüften zB ist eine Aktion in welcher ich der WinMatic einen Einschaltbefehl mit gleichzeitig aktiviertem ON_TIME mitschicke, sich das Gerät also selbständig nach Ablauf dieser Zeit (auch bei Kommunikationsstörungen etc) abschaltet/schließt. 
Selbiges gilt für unsere Bewässerung. 
Dieses Mitschicken der ON_TIME Option ist eigentlich einer der Hauptgründe für die Virtuellen Tasten und hat sich bewährt.
Wir sitzen somit nicht herum und drücken Knöpfe :) (das wäre ja auch bezüglich einer -automatisierung kontraproduktiv)

**Reply by danielperna84** [reply]
Ach so. Also geht es dir gar nicht direkt um die Knöpfe im UI (das war hier ja auch nebenher Thema), sondern nur den Service um die Taste per Automation zu drücken. Das mit dem `ON_TIME` mache ich selbst auch mit so einem virtuellen Taster (in meiner Prod-Umgebung mit der alten Integration). Da kann ich also bestätigen, dass man das auch aus Skripten oder sonstwo heraus können sollte.

**Comment by AlexaMiller** [comment]
Doch, ich hätte bitte auch gerne die Knöpfe in der UI, denn zB um das Garagentor, die Einfahrt West, das Gartentürl oder auch die Dachbodenstiege zu öffnen .. das sollte auch über das User Interface funktionieren. 
Ja, meist eh vollautomatisiert, aber manchmal möchte ich das halt manuell aufrufen.

Und manchmal braucht auch einer der Olivenbäume mehr Wasser, also auch die Bewässerung.

**Comment by AlexaMiller** [comment, marked as answer]
Soderle, 
Problem ist gelöst - funktioniert wunderbar indem ich einfach das Gerät selbst aufrufe:

service: hahm.set_device_value
data:
  device_id: <device_id_der_HmIP-RCV-50>
  parameter: PRESS_SHORT
  value: 'true'
  value_type: boolean
  channel: 3

Vielen lieben Dank, jetzt noch einen Kaffee und dann in den wohlverdienten Feierabend!
